### DISCUSSION TITLE
Using lists as parameters

### DISCUSSION THREAD

**Discussion by StephanSST** [Dynatrace/dynatrace-configuration-as-code#494] [Q&A] [answered]
Hi there
Thanks for providing Monaco. I think its a great tool that helps us syncing configurations over environments. It also shows us where we can optimize configurationes. So great job, please keep on doing it!

Right now we have an issue with email notifications: The receivers of a notification is a List of Strings (string[]) on the API. We would like to parametrize them but we failed. How is the correct syntax for lists? 

The json looks like this:
```{
 "name": "{{.name}}",
 "active": true,
 "alertingProfile": "{{.alertingProfileId}}",
 "body": "{{.body}}",
 "receivers": [
  "{{.receiver}}"
 ],
 "subject": "{{.subject}}",
 "type": "EMAIL"
}
```

But how can I add them in the notfication.yaml. I failed with these examples:
```
- receiver: "personA@mail.ch" two receivers splitted on two lines, so together with seconds line
- receiver: "personA@mail.ch" -> only adding second email
- receiver: "personA@mail.ch", "personA@mail.ch" -> Failed to unmarshal yaml
- receiver: "{personA@mail.ch, personA@mail.ch}" -> Failed to unmarshal yaml
- receiver: "[personA@mail.ch, personA@mail.ch]" -> Failed to unmarshal yaml
- receiver: 
    - "personA@mail.ch"
    - "personB@mail.ch" -> notification.yaml could not be parsed: cannot convert YAML
- receiver: "[ "personA@mail.ch", "personB@mail.ch" ]" -> Failed to unmarshal yaml
- receiver: [ "personA@mail.ch", "personB@mail.ch" ] -> notification.yaml could not be parsed: cannot convert YAML
```

Is it possible using lists as parameters? Did anybody ever tried this?

Kind regards and thanks for trying to help!
Stephan

**Comment by didiladi** [comment]
Hi @StephanSST! Thanks for the kind words about monaco :slightly_smiling_face: 

We currently just support plain strings passed in the template. Try the following in the yaml (wrapping the list of all addresses in single quotes(`'`)):

```
email:
  - receiver: '"personA@mail.ch", "personB@mail.ch"'
```

**Comment by StephanSST** [comment, marked as answer]
Hi Dieter
I tried your tip but it's not working. I still get an error:
```
2021-11-29 09:34:24 ERROR       File did not contain valid json:
 --> .\template\notification\EMail.json:7:6
   |  "receivers": [
 7 |   " "personA@mail.ch", "personB@mail.ch" "
   |      ^^^
   - Cause: invalid character 'p' after array element
2021-11-29 09:34:24 ERROR Errors during deployment! Check log!
```

But with your tip and the error message I assumed that the double quote (`"`) sign at the beginning and at the end is constant so I just removed them and tried this new option:

`- receiver: personA@mail.ch", "personB@mail.ch`

and it worked. It looks pretty strange but now I get two email receivers in my alert notification. So thank you very much for your fast reply. Your answer brought me to the right path.

Kind regards,
Stephan

**Reply by didiladi** [reply]
Hi, yes, you are right. This is one option - I just checked again. We have the following in our json:

```
    "receivers": [
        {{ .receivers }}
    ],
```

We omit the double quote (`"`) in the json - then it works as well.

You could also try to combine the two methods:

```
yaml:
- receiver: "personA@mail.ch", "personB@mail.ch"

json:
    "receivers": [
        {{ .receivers }}
    ],
```
